### PR TITLE
EWM7751 replaced save tab of reduction with pop up, removed some noisy warnings

### PR DESCRIPTION
**Pull request by walshmm**
## Description of work

This exposes the workflow completion message to the `WorkflowBuilder` to enable an implemented workflow to display a custom message upon completion.
This was then utilized to replace the Save step of the `ReductionWorkflow` (which does nothing) with a simple pop up that accomplishes the same goal: Informing the user that redcution has completed successfully.

This turns off a particularly noisy in-actionable warning message too.

## To test

Run Reduction with run number `59039`, observe that it completes.
Ensure the reduction results are in the workspace list and are also persisted to disk. 

## Link to EWM item

[EWM#7751](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=7751)

### Verification

- [x] the author has read the EWM story and acceptance critera
- [x] the reviewer has read the EWM story and acceptance criteria
- [x] the reviewer certifies the acceptance criteria below reflect the criteria in EWM

### Acceptance Criteria

This list is for ease of reference, and does not replace reading the EWM story as part of the review.  Verify this list matches the EWM story before reviewing.

- [x] The new desired solution is to skip the final step (The Reduction Save View) on the reduction workflow and instead simply drop an a prompt informing the user that Reduction had completed successfully
- [x] The user no longer has the opportunity to create workspaces that will accidentally be deleted at the end of the Reduction Workflow (i.e. painstakingly created masks)

### CODE REVIEW

File: src/snapred/ui/workflow/ReductionWorkflow.py [hunk 1]
```diff
@@ -50,7 +53,7 @@ def __init__(self, parent=None):
                 "Reduction",
                 continueAnywayHandler=self._continueAnywayHandler,
             )
-            .addNode(self._nothing, self._reductionSaveView, "Save")
+            # .addNode(self._nothing, self._reductionSaveView, "Save")
```
Comment (dlcaballero16) [inline src/snapred/ui/workflow/ReductionWorkflow.py:56]: I would delete this line entirely, or add a comment that this will be cleaned up later

Comment (walshmm) [reply src/snapred/ui/workflow/ReductionWorkflow.py]: done, also updated the completion text

File: src/snapred/ui/workflow/WorkflowBuilder.py [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
-    def __init__(self, *, startLambda=None, iterateLambda=None, resetLambda=None, cancelLambda=None, parent=None):
+    def __init__(
+        self,
+        *,
+        startLambda=None,
+        iterateLambda=None,
+        resetLambda=None,
+        cancelLambda=None,
+        parent=None,
+        completionMessageLambda=lambda: Config["ui.default.workflow.completionMessage"],
```
Comment (walshmm) [inline src/snapred/ui/workflow/WorkflowBuilder.py:15]: I feel like I could reduce the number of lambdas by just passing the WorkflowImplementer itself.

**Review by dlcaballero16 — APPROVED** [review]
Tested and reduction works as it should.

**Pull request merged**